test/extended/apiserver/disruption.go

```diff
@@ -0,0 +1,496 @@
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callWorkerDaemonset(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	err = callRBACClusterAdmin(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callRBACHostaccess(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callServiceAccount(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callProject(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+func createDaemonset(ctx context.Context, oc *exutil.CLI, create bool, obj *appsv1.DaemonSet) error {
+	client := oc.AdminKubeClient().AppsV1().DaemonSets(namespace)
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		if err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+			ds, err := client.Get(context.Background(), obj.Name, metav1.GetOptions{})
+			if err != nil {
+				framework.Logf("error getting daemonsets %v", err)
+				return false, nil
+			}
+			return ds.Status.NumberReady > 0, nil
+		}); err != nil {
+			return fmt.Errorf("daemonset %s didn't roll out: %v", obj.Name, err)
+		}
+
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+		if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+			return nil
+		}
+	}
+
+	return err
+}
+
+func callMasterDaemonset(ctx context.Context, oc *exutil.CLI, create bool, apiIntHost string) error {
+	labels := map[string]string{
+		"app": "pod-monitor-master",
+	}
+	name := "pod-monitor-masters"
+	truePointer := true
+	obj := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "artifacts",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: disruptionDataPath,
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "internal-lb",
+							Image: "quay.io/vrutkovs/ocp:tests-92de0978f8",
+							Env: []corev1.EnvVar{
+								{
+									Name:  "KUBERNETES_SERVICE_HOST",
+									Value: apiIntHost,
+								},
+								{
+									Name:  "KUBERNETES_SERVICE_PORT",
+									Value: "6443",
+								},
+								{
+									Name: "NODE_NAME",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "spec.nodeName",
+										},
+									},
+								},
+							},
+							Command: []string{
+								"openshift-tests",
+								"run-monitor",
+								"--api-disruption-only",
+								"--extra-locator",
+								"target=api-int",
```

Comment:
I think we need you to get this into the BackendName, not the locator. This is effectively hitting a different backend, this will jive with what Abu is doing as well. If we don't, all the disruption will get lumped together in the db, which is definitely not what we want, we need this distinct and tracked separately. I would look to the naming scheme evolving in Abu's PR for example: backend-sampler-openshift-apiserver-http2-external-lb-reused-connections. Essentially if you're re-testing what was previously kube-api-new-connections, your variant should be kube-api-int-lb-new-connections, or kube-api-service-network-new-connections. (etc)

Reply:
Yes the db records disruption for the run for a given backend, not per host. You'd need a way to alter the backend names in your use case so there'd be a refactor needed there. I'm not sure if Abu's PR contains anything to make that easier but yes we can hold this for sure if there is.